src/constant.jl

```diff
@@ -5,54 +5,47 @@
 export Constant
 export vexity, evaluate, sign, conic_form!
 
-struct Constant <: AbstractExpr
+const ComplexValue = Union{Complex,AbstractVecOrMat{<:Complex}}
+
+_size(x::Number) = (1, 1)
+_size(x::AbstractVector) = (length(x), 1)
+_size(x::AbstractMatrix) = size(x)
+
+_sign(x::ComplexValue) = ComplexSign()
+_sign(x::Number) = x >= 0 ? Positive() : Negative()
+_sign(v::AbstractVecOrMat) = all(x->x >= 0, v) ? Positive() : Negative()
```

Comment:
I just realized this isn't the same behavior as before. On master, when the sign is detected, it's only `Negative()` if _all_ values are less than or equal to 0. Currently in this PR, I'm calling it negative if any are not greater than or equal to 0, which is wrong.